src/update-promoted-values.ts

```diff
@@ -31,27 +59,77 @@ export async function updatePromotedValues(
   // If the file is empty (or just whitespace or whatever), that's fine; we
   // can just leave it alone.
   if (!document) {
-    return contents;
+    return [contents, new Map()];
   }
 
   // We decide what to do and then we do it, just in case there are any
   // overlaps between our reads and writes.
   logger.info('Looking for promote');
-  const promotes = findPromotes(document, promotionTargetRE2);
+  const promotes = await findPromotes(
+    document,
+    promotionTargetRE2,
+    dockerRegistryClient,
+    gitHubClient,
+  );
+
+  logger.info(`Promotes: ${JSON.stringify(promotes)}`);
 
   logger.info('Copying values');
   for (const { scalarTokenWriter, value } of promotes) {
     scalarTokenWriter.write(value);
   }
-  return stringify();
+
+  const relevantCommits: Map<string, RelevantCommit[]> = new Map();
+  for (const [serviceName, commits] of promotes.map((p) => p.relevantCommits)) {
+    relevantCommits.set(serviceName, commits);
+  }
+  return [stringify(), relevantCommits];
 }
 
-function findPromotes(
+async function findPromotes(
   document: yaml.Document.Parsed,
   promotionTargetRE2: RE2 | null,
-): Promote[] {
+  dockerRegistryClient: DockerRegistryClient | null,
+  gitHubClient: GitHubClient | null = null,
+): Promise<Promote[]> {
   const { blocks } = getTopLevelBlocks(document);
   const promotes: Promote[] = [];
+
+  //
+  // Expected format of a block:
+  //
+  // my-service-prod:
+  //   track: <branch (main) | pr (pr-1234)>
+  //   gitConfig:
+  //     ref: <commit>
+  //   dockerImage:
+  //     tag: main---0013586-2024.04-<commit>
+  //   promote:
+  //     from: my-service-staging
+  //
+  //
+  // Expected format of global:
+  //
+  // global:
+  //   datadogServiceName: my-service
+  //   gitConfig:
+  //     repoURL: https://github.com/owner/repo.git
+  //     path: k8s/services/service
+  //   dockerImage:
+  //     repository: service
+  //
+  const repoURL: string | undefined = document.getIn([
```

Comment:
Both this and dockerImage.repository below can be set in the individual app block in addition to in `global` (individual overrides global) as implemented in the findTrackables functions in the other files.